pkg/products/threescale/reconciler.go

```diff
@@ -261,7 +276,33 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
-func (r *Reconciler) reconcileTemplates(ctx context.Context, installation *integreatlyv1alpha1.RHMI, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+// restores seed and master api cast secrets if available
+func (r *Reconciler) restoreSystemSecrets(ctx context.Context, serverClient k8sclient.Client, installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	for _, secretName := range []string{systemSeedSecretName, systemMasterApiCastSecretName} {
+		err := resources.CopySecret(ctx, serverClient, secretName, installation.Namespace, secretName, r.Config.GetNamespace())
+		if err != nil {
+			if !k8serr.IsNotFound(err) {
+				return integreatlyv1alpha1.PhaseFailed, err
```

Comment:
is failed the best state here? The secret could already exist and the error be really quite trivial, right?

Reply:
Good question. If the secret already exists in 3scale then it would get updated. `PhaseFailed` here means that the integreatly operator can't reconcile (restore) the 3scale secrets. But we should probably also ignore conflicts because the 3scale operator might also update those secrets.

Reply:
I think it's still best to be able to see those conflict errors rather than ignore them because then we won't know that the restore has actually failed. 

If we get any errors back from `CopySecret`, the secret wouldn't have been updated with the original credentials right? I think we should capture those errors as an event.

Reply:
ok, I'm happy for this to merge.